### ISSUE TITLE
PATHS showing more than once when running ide env

### ISSUE THREAD

**Issue by aBega2000** [devonfw/IDEasy#323]
### Expected behavior

As a user, I want the PATHS to be shown only once when I run ide env

### Actual behavior

Some Paths are shown more than once

### Steps to reproduce (bug) / Use Case of feature request (enhancement)

1. open git bash terminal
2. run "ide env"

![image](https://github.com/devonfw/IDEasy/assets/145266540/8b0c1b3a-df1c-4a11-b61f-ca790e28810b)
This Image shows some of the repeated PATHs

**Comment by hohwille** [comment]
I had already analyzed this bug and found the root problem and the solution that I posted in our chat but so it wont get lost, I will add it here:
https://github.com/devonfw/IDEasy/blob/0e8f59f1a68cceb2fa9f4f5f3593d0d09bcc5ae8/cli/src/main/java/com/devonfw/tools/ide/context/AbstractIdeContext.java#L264
Method `setCwd` is called from here:
https://github.com/devonfw/IDEasy/blob/0e8f59f1a68cceb2fa9f4f5f3593d0d09bcc5ae8/cli/src/main/java/com/devonfw/tools/ide/context/AbstractIdeContext.java#L176
But `ideRoot` is set only after that:
https://github.com/devonfw/IDEasy/blob/0e8f59f1a68cceb2fa9f4f5f3593d0d09bcc5ae8/cli/src/main/java/com/devonfw/tools/ide/context/AbstractIdeContext.java#L200
Hence `ideRoot` is `null` and therefore the `getTool` method cannot detect that something is a tool path:
https://github.com/devonfw/IDEasy/blob/0e8f59f1a68cceb2fa9f4f5f3593d0d09bcc5ae8/cli/src/main/java/com/devonfw/tools/ide/common/SystemPath.java#L99-L102

**Comment by hohwille** [comment]
BTW: Another question would be what should happen now if I go outside of an IDEasy project and just type `ide`.
It currently gives me this
```
$ ide
IDE_ROOT is not set or not a valid directory.
```
Expected would be something like this:
```
$ ide
You are not inside an IDEasy installation: /d/projects
```
(please create a new bug issue for that)
But the actual question is if after calling `devon` the `PATH` should be "resetted" so that all IDEasy tools should now be removed from the PATH again or not?

With `devon` it does not do this... So to remove the tools from the `PATH` I would simply have to open a new shell?

```
hohwille@CE49454 MSYS /d/projects
$ echo $PATH
/c/Users/hohwille/bin:/usr/local/bin:/usr/bin:/bin:/opt/bin:/d/oracle-xe/product/18.0.0/dbhomeXE/bin:/c/WINDOWS/system32:/c/WINDOWS:/c/WINDOWS/System32/Wbem:/c/WINDOWS/System32/WindowsPowerShell/v1.0:/c/WINDOWS/System32/OpenSSH:/c/Program Files (x86)/GnuPG/bin:/c/Program Files/SlikSvn/bin:/c/Program Files/TortoiseSVN/bin:/cmd:/c/Program Files/Rancher Desktop/resources/resources/win32/bin:/c/Program Files/Rancher Desktop/resources/resources/linux/bin:/c/Users/hohwille/scripts:/d/projects/_ide/bin:/usr/bin/vendor_perl:/usr/bin/core_perl

hohwille@CE49454 MSYS /d/projects
$ cd devonfw

hohwille@CE49454 MSYS /d/projects/devonfw
$ devon
devonfw-ide environment variables have been set for /d/projects/devonfw in workspace main

hohwille@CE49454 MSYS /d/projects/devonfw
$ echo $PATH
/d/projects/devonfw/software/vscode/bin:/d/projects/devonfw/software/tomcat/bin:/d/projects/devonfw/software/terraform:/d/projects/devonfw/software/sonar/bin:/d/projects/devonfw/software/python:/d/projects/devonfw/software/pip:/d/projects/devonfw/software/oc:/d/projects/devonfw/software/node_modules:/d/projects/devonfw/software/node:/d/projects/devonfw/software/mvn/bin:/d/projects/devonfw/software/lazydocker:/d/projects/devonfw/software/kotlinc/bin:/d/projects/devonfw/software/java/bin:/d/projects/devonfw/software/intellij/bin:/d/projects/devonfw/software/helm:/d/projects/devonfw/software/gradle/bin:/d/projects/devonfw/software/gh/bin:/d/projects/devonfw/software/gcviewer:/d/projects/devonfw/software/gcloud/bin:/d/projects/devonfw/software/eclipse:/d/projects/devonfw/software/dotnet:/d/projects/devonfw/software/cobigen/bin:/d/projects/devonfw/software/az/bin:/d/projects/devonfw/software/aws:/d/projects/devonfw/software/android-studio/bin:/c/Users/hohwille/bin:/usr/local/bin:/usr/bin:/bin:/opt/bin:/d/oracle-xe/product/18.0.0/dbhomeXE/bin:/c/WINDOWS/system32:/c/WINDOWS:/c/WINDOWS/System32/Wbem:/c/WINDOWS/System32/WindowsPowerShell/v1.0:/c/WINDOWS/System32/OpenSSH:/c/Program Files (x86)/GnuPG/bin:/c/Program Files/SlikSvn/bin:/c/Program Files/TortoiseSVN/bin:/cmd:/c/Program Files/Rancher Desktop/resources/resources/win32/bin:/c/Program Files/Rancher Desktop/resources/resources/linux/bin:/c/Users/hohwille/scripts:/d/projects/_ide/bin:/usr/bin/vendor_perl:/usr/bin/core_perl

hohwille@CE49454 MSYS /d/projects/devonfw
$ cd ..

hohwille@CE49454 MSYS /d/projects
$ devon
You are not inside a devonfw-ide installation: /d/projects

hohwille@CE49454 MSYS /d/projects
$ echo $PATH
/d/projects/devonfw/software/vscode/bin:/d/projects/devonfw/software/tomcat/bin:/d/projects/devonfw/software/terraform:/d/projects/devonfw/software/sonar/bin:/d/projects/devonfw/software/python:/d/projects/devonfw/software/pip:/d/projects/devonfw/software/oc:/d/projects/devonfw/software/node_modules:/d/projects/devonfw/software/node:/d/projects/devonfw/software/mvn/bin:/d/projects/devonfw/software/lazydocker:/d/projects/devonfw/software/kotlinc/bin:/d/projects/devonfw/software/java/bin:/d/projects/devonfw/software/intellij/bin:/d/projects/devonfw/software/helm:/d/projects/devonfw/software/gradle/bin:/d/projects/devonfw/software/gh/bin:/d/projects/devonfw/software/gcviewer:/d/projects/devonfw/software/gcloud/bin:/d/projects/devonfw/software/eclipse:/d/projects/devonfw/software/dotnet:/d/projects/devonfw/software/cobigen/bin:/d/projects/devonfw/software/az/bin:/d/projects/devonfw/software/aws:/d/projects/devonfw/software/android-studio/bin:/c/Users/hohwille/bin:/usr/local/bin:/usr/bin:/bin:/opt/bin:/d/oracle-xe/product/18.0.0/dbhomeXE/bin:/c/WINDOWS/system32:/c/WINDOWS:/c/WINDOWS/System32/Wbem:/c/WINDOWS/System32/WindowsPowerShell/v1.0:/c/WINDOWS/System32/OpenSSH:/c/Program Files (x86)/GnuPG/bin:/c/Program Files/SlikSvn/bin:/c/Program Files/TortoiseSVN/bin:/cmd:/c/Program Files/Rancher Desktop/resources/resources/win32/bin:/c/Program Files/Rancher Desktop/resources/resources/linux/bin:/c/Users/hohwille/scripts:/d/projects/_ide/bin:/usr/bin/vendor_perl:/usr/bin/core_perl
```

**Comment by mvomiero** [comment]
Now that we have:
* #401 PR to remove the duplicates
* #400 Issue for the output of ide outside of an IDEasy installation

Should we create another issue for the following topic, so we have time to think about the best approach to handle it, or we decide directly in the following team meeting? 
**_"But the actual question is if after calling devon the PATH should be "resetted" so that all IDEasy tools should now be removed from the PATH again or not?"_**. See comment above.
